tools/gulp/tasks/breaking-changes.ts

```diff
@@ -0,0 +1,108 @@
+import {task} from 'gulp';
+import {join, relative} from 'path';
+import {readFileSync} from 'fs';
+import {bold, red, green} from 'chalk';
+import * as ts from 'typescript';
+import * as tsutils from 'tsutils';
+
+// Current version from the package.json. Splits it on the dash to ignore `-beta.x` suffixes.
+const packageVersion = require(join(process.cwd(), 'package.json')).version.split('-')[0];
+
+// Regex used to extract versions from a string.
+const versionRegex = /\d+\.\d+\.\d+/;
+
+/**
+ * Goes through all of the TypeScript files in the project and puts
+ * together a summary of all of the pending and expired breaking changes.
+ */
+task('breaking-changes', () => {
+  const cwd = process.cwd();
+  const configFile = ts.readJsonConfigFile(join(cwd, 'tsconfig.json'), ts.sys.readFile);
```

Comment:
Would it be better to just refer to the build config's project directory? 

<sup>I know that gulp always changes the CWD to the `gulpfile` directory</sup>

Reply:
Yeah I'd say so, I didn't know that we had it configured somewhere.